homeassistant/components/sensor/ambient_station.py

```diff
@@ -0,0 +1,205 @@
+        """Initialize the sensor."""
+        self._state = None
+        self.station_data = station_data
+        self._condition = condition
+        self._name = name
+        self._units = units
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._units
+
+    def update(self):
+        """Fetch new state data for the sensor.
+
+        This is the only method that should fetch new data for Home Assistant.
+        """
+        _LOGGER.debug("Getting data for sensor: %s", self._name)
+        data = self.station_data.get_data()
+        if data is None:
+            # update likely got throttled and returned None, so use the cached
+            # data from the station_data object
+            self._state = self.station_data.data[self._condition]
+        else:
+            if self._condition in data:
+                self._state = data[self._condition]
+            else:
+                _LOGGER.warning("%s sensor data not available from the "
+                                "station", self._condition)
+
+        _LOGGER.debug("Sensor: %s | Data: %s", self._name, self._state)
+
+
+class AmbientStationData:
+    """Class to interface with ambient-api library."""
+
+    def __init__(self, api_key, app_key, interval):
+        """Initialize station data object."""
+        self._api_key = api_key
+        self._app_key = app_key
+
+        self._interval = interval
+
+        self._api_keys = {
+            'AMBIENT_ENDPOINT':
+            'https://api.ambientweather.net/v1',
+            'AMBIENT_API_KEY': self._api_key,
+            'AMBIENT_APPLICATION_KEY': self._app_key,
+            'log_level': 'DEBUG'
+        }
+
+        self.data = None
+        self._station = None
+        self._api = None
+        self._devices = None
+
+        self.get_data = Throttle(self._interval)(self.update)
+
+    def update(self):
+        """Get new data."""
+        # refresh API connection since servers turn over nightly
+        _LOGGER.debug("Getting new data from server")
+        new_data = None
+        self._connect_api()
+        time.sleep(2)   # need minimum 2 seconds between API calls
+        if self._station is not None:
+            data = self._station.get_data()
+            if data is not None:
+                new_data = data[0]
+                self.data = new_data
+            else:
+                _LOGGER.debug("data is None type")
+        else:
+            _LOGGER.debug("Station is None type")
+
+        return new_data
+
+    def _connect_api(self):
+        """Connect to the API and capture new data."""
+        from ambient_api.ambientapi import AmbientAPI
+
+        self._api = AmbientAPI(**self._api_keys)
```

Comment:
Why do we need to create a new instance every update? Can't we reuse the same api instance?

Reply:
See previous comment above.  Servers turn over every day and you need to reconnect to refresh the connection otherwise you're data requests will get blocked.  It's unknown when the servers turn over so refreshing the connection every update seemed like the only reliable solution.